go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1986,35 +1987,45 @@ func commonFlows(subnets []*net.IPNet, bridge *bridgeConfiguration, isPodNetwork
 	}
 
 	if ofPortPhys != "" {
-		if config.Gateway.DisableSNATMultipleGWs || isPodNetworkAdvertised {
+		if config.Gateway.DisableSNATMultipleGWs || isPodNetworkAdvertised || isUDNNetworkAdvertised {
```

Comment:
We shouldn't use these `PodNetworkAdvertised`,`isUDNNetworkAdvertised` any more. That information comes now from `bridge.netConfig`. Yoyu shoudl just iterate that and based the config from information obtained there.